util/addon.c

```diff
@@ -0,0 +1,47 @@
+#include <stdlib.h>
+#include <wayland-server-core.h>
+
+#include "wlr/util/addon.h"
+
+void wlr_addon_set_init(struct wlr_addon_set *set) {
+	wl_list_init(&set->addons);
+}
+
+void wlr_addon_set_finish(struct wlr_addon_set *set) {
+	struct wlr_addon *addon, *tmp;
+	wl_list_for_each_safe(addon, tmp, &set->addons, link) {
+		wlr_addon_destroy(addon);
+	}
+}
+
+struct wlr_addon *wlr_addon_create(struct wlr_addon_set *set, void *data,
+		void *owner, wlr_addon_finish_data_func_t finish_data) {
```

Comment:
Maybe we should assert that we don't have another addon with the same owner?

Reply:
Good idea.